src/installation/guides/chroot.md

```diff
@@ -0,0 +1,298 @@
+System Partition
+
+```
+# mount /dev/sda2 /mnt/
+# mkdir -p /mnt/boot/efi/
+# mount /dev/sda1 /mnt/boot/efi/
+```
+
+Initialize swap space, if desired, using
+[mkswap(8)](https://man.voidlinux.org/mkswap.8).
+
+## Base Installation
+
+> Note: this step is separated into the XBPS method and the ROOTFS method.
+> Follow only one of the two following subsections.
+
+### The XBPS Method
+
+[Select an XBPS mirror.](../../xbps/repositories/mirrors/index.md) For
+simplicity, save this URL to a shell variable for later use. Append `/current`
+to this repository url to install a glibc distribution, append `/current/musl`
+for a musl distribution.
+
+> Note: musl binaries are not available for i686.
+
+For example, this is the url for glibc from the Tier 1 server in Finland,
+
+```
+# REPO=https://alpha.de.repo.voidlinux.org/current
+```
+
+xbps also needs to know what architecture is being installed, available options
+are `x86_64`, `x86_64-musl` and `i686` for PC architecture computers.
+
+For example,
+
+```
+# ARCH=x86_64
+```
+
+This architecture must be compatible with your current operating system, but
+does not need to be the same. If your host is running an x86_64 operating
+system, any of the three architectures can be installed (whether the host is
+musl or glibc), but an i686 host can only install i686 distributions.
+
+Use [xbps-install](https://man.voidlinux.org/xbps-install.1) to boot strap the
+distribution by installing the `base-system` metapackage.
+
+```
+# XBPS_ARCH=$ARCH xbps-install -S -r /mnt -R "$REPO" base-system
+```
+
+### The ROOTFS Method
+
+[Download a ROOTFS tarball](https://alpha.de.repo.voidlinux.org/live/current/)
+matching your architecture.
+
+Unpack the tarball into the newly configured filesystems.
+
+```
+# tar xvf void-[...]-ROOTFS.tar.xz -C /mnt
+```
+
+## Configuration
+
+### Entering the Chroot
+
+Mount the pseudo-filesystems needed for a chroot,
+
+```
+# mount -t proc /proc /mnt/proc
+# mount -t sysfs /sys /mnt/sys
+# mount -o bind /dev /mnt/dev
+# mount -t devpts pts /mnt/dev/pts
+```
+
+Copy the DNS configuration into the new root so that xbps can still download new
+packages inside the chroot,
+
+```
+# cp /etc/resolv.conf /mnt/etc/
+```
+
+Chroot into the new installation
+
+```
+# PS1='(chroot) # ' chroot /mnt/ /bin/bash
+```
+
+### Update the Distribution (ROOTFS method only)
+
+> Note: This section should be skipped if following the XBPS method. Your
+> distribution is already up to date.
+
+ROOTFS images generally contain out of date software and do not come with a
+complete `base-system`. Update the distribution and install `base-system`.
+
+```
+# xbps-install -Su
+# xbps-install base-system
+```
+
+> Note: the first step will likely only update xbps. Running `xbps-install -u` a
+> second time will update all other packages. However, these packages will all
+> be updated when `base-system` is installed so that step can be safely skipped.
```

Comment:
> I prefer

For a general introductory guide, though?

Way back at the beginning this guide included a brief discussion on `base-system`, `base-voidstrap` and `base-minimal`. It was decided that this guide should only cover `base-system`. That was before any mention of installing from a ROOTFS tarball, which (iirc) contains `base-voidstrap`. I include installing `base-system` so that you end up in the same place you would have if you were installing from xbps.

> installing more stuff.

I think this is the big issue. What is more stuff? At the very least, you would have to install a kernel. Then there are a number of tools you don't get with base-voidstrap that new users might expect they would have installed because other documentation assumes base-system is installed. Having some documentation about building more minimal installations without relying on base-system would be good, but I don't think this is the place for it.

Not installing base-system I think is a more advanced thing to do. I'm not sure if this is official policy, but some xbps packages depend on software that gets installed through base-system but do not explicitly list it as a dependency. cryptsetup is in this boat. If you installed it at this point in the procedure it would not work because it depends on glib.

Reply:
The rootfs does use `base-voidstrap`. If you advocate switching to `base-system`, I recommend removing `base-voidstrap` to avoid "Why do I have two base packages?" questions that may arise.

In addition to a kernel, most systems will probably require one or more firmware packages be installed if a specific kernel version is installed rather than the `linux` meta-package. I agree that moving to `base-system` is probably good general advice. Maybe a note below this can point out the option of skipping `base-system` and the need to install at least some of these dependants.

Reply:
My point was more that I prefer to depend on a full update, rather on than the installation of `base-system` necessarily updating everything.

ahesford has a point, though. We should probably explain that ROOTFS requires some more packages "such as a kernel...". Then mention that you can just install the base-system package to get all the necessary goodies.

Reply:
ah, I see what you're saying.

My problem with `xbps-install -u ; xbps-install base-system` is that it's a pain on a slow internet connection, where you want to start the installation and walk away. Breaking this step up means that you have to come back half way through the process and set the next thing up.

How about a quick section at the end (before rebooting) instructing the reader to `xbps-install -Su`? At this stage, nothing should happen (unless there happened to be updates in that time) but it guarantees that *everything* is up to date, and it's a useful reminder to new users how to do a full system update.

> ahesford has a point, though. We should probably explain that ROOTFS requires some more packages "such as a kernel...". Then mention that you can just install the base-system package to get all the necessary goodies.

This exact section was removed after discussing it with @bobertlo and @Duncaen, so I'm not going to put it back in without their input.

> I recommend removing base-voidstrap

Yes, I'll do that.

Reply:
> Breaking this step up means that you have to come back half way through the process and set the next thing up.

You can recommend the `-y` flag for that situation, if you think it's worth adding. I'd mention the possibility of using it, but the commands should probably be left as they are.

> How about a quick section at the end (before rebooting) instructing the reader to xbps-install -Su

I think this is definitely worth it, but I'd prefer if you just linked to the update section instead. Kind of like how the live image installation guide currently does it.

> This exact section was removed after discussing it with @bobertlo and @Duncaen, so I'm not going to put it back in without their input.

Fair enough.

Reply:
> You can recommend the -y flag for that situation, if you think it's worth adding

I don't think it's worth mentioning. I'm just not seeing any advantage to updating the whole system before installing base-system

Reply:
okay, I did some more testing and installing base-system doesn't actually do a full system update so this is definitely going in.